lua/nvim-drawer/init.lua

```diff
@@ -199,6 +199,12 @@ function mod.create_drawer(opts)
       bufnr = instance.state.windows_and_buffers[winid] or -1
     end
 
+    -- ... or if we do have a window but don't have a buffer, we use the buffer
+    -- of the window.
+    if winid ~= -1 and bufnr == -1 then
+      bufnr = vim.api.nvim_win_get_buf(winid)
+    end
```

Comment:
This feels "technically correct": when `.open` is called, and _somehow_ there is a window that belongs to the drawer but it _doesn't_ contain a known buffer, that buffer is used.

The controversial / potential bugs bit is the fact that it happens after the logic for `should_reuse_previous_bufnr` happens, and that logic _very intentionally_ defaults back to -1.